packages/katello/katello/hostname-change.rb

```diff
@@ -551,7 +551,9 @@ def run
           STDOUT.puts result
           STDOUT.puts 'Restarting puppet services'
           run_cmd('/bin/systemctl try-restart puppet')
-          run_cmd('foreman-maintain service restart --only puppetserver')
+          if File.exist?('/usr/lib/systemd/system/puppetserver.service')
+            run_cmd('foreman-maintain service restart --only puppetserver')
```

Comment:
Why do we shell out to `foreman-maintain`? My reasoning when reviewing it back then is that I thought it would not restart it if it wasn't enabled. If it doesn't provide this guarantee then this is really the same as:
```suggestion
            run_cmd('systemctl restart puppetserver')
```

And again, getting into the territory of "why am I looking at this": Shouldn't we really restart the server before the client?

Reply:
@upadhyeammit says this should exist 0 and carry on in this case, so probably this patch is not needed when we fix the issue in FM, nice

Reply:
so the FM fix is in https://projects.theforeman.org/issues/34100, but after applying this, FM does *not* exit 0 in the case where we ask it to restart a service that isn't there.

Reply:
Since there was some confusion, I explained what my original reasoning was. In https://github.com/theforeman/foreman-packaging/pull/4249 when I last looked at this I don't see comments, but I recall having offline conversations about why we need a wrapper at all. Also, back then `puppetserver` wasn't really easily optional.

> And again, getting into the territory of "why am I looking at this": Shouldn't we really restart the server before the client?

Looks like I am repeating myself: https://github.com/theforeman/foreman-packaging/pull/4249#discussion_r340711477

In short: If we do use `foreman-maintain`, then it should be powerful enough to not need any manual workarounds but I prefer to purely use `systemctl` and not `foreman-maintain` here.

Reply:
flipped to systemctl